### ISSUE TITLE
twoBitToFa error

### ISSUE THREAD

**Issue by BoBo2021jenny** [hillerlab/CESAR2.0#21]
Hi,

I am trying to run"annotateGenesViaCESAR.pl ....", but the following error is :

Argument "" isn't numeric in numeric gt (>) at /data/zhangbo/software/CESAR2.0-master/tools/annotateGenesViaCESAR.pl line 329.
Argument "" isn't numeric in numeric gt (>) at /data/zhangbo/software/CESAR2.0-master/tools/annotateGenesViaCESAR.pl line 329.
Argument "" isn't numeric in numeric gt (>) at /data/zhangbo/software/CESAR2.0-master/tools/annotateGenesViaCESAR.pl line 329.
Argument "" isn't numeric in numeric gt (>) at /data/zhangbo/software/CESAR2.0-master/tools/annotateGenesViaCESAR.pl line 329.
Argument "" isn't numeric in numeric gt (>) at /data/zhangbo/software/CESAR2.0-master/tools/annotateGenesViaCESAR.pl line 329.
file /dev/shm/species.bed.0aHTuQ doesn't appear to be in bed format. At least 4 fields required, got 3
Error running twoBitToFa call for species "XXX"

Thanks for the reply!

**Comment by MichaelHiller** [comment]
If you send me the input files and the exact command you used, I can try to have a look if I find the problem. 

Alternatively, pls try https://github.com/hillerlab/TOGA which integrates CESAR and has many additional features (orthology, gene losses, etc)

**Comment by BoBo2021jenny** [comment]
Hi
Thank you for your reply!
"CRITICAL src/Viterbi.c:240 Viterbi():	No observations.
Error running cesar on exon, cesar /dev/shm/cesarIn.fa.BKLjUP -f -l --matrix......" is in the output log. What is the reason for this?

**Comment by MichaelHiller** [comment]
Are you running TOGA or CESAR? 
And can you pls send the log? 
Does /dev/shm/ exist on your machine?

**Comment by BoBo2021jenny** [comment]
I am running  CESAR. /dev/shm/ exists on my machine. The log is "CRITICAL src/Viterbi.c:240 Viterbi():	No observations.
Error running cesar on exon, '/software/CESAR2.0-master/cesar /dev/shm/cesarIn.fa.BKLjUP -f -l --matrix /software/CESAR2.0-master/extra/tables/human/eth_codon_sub.txt -p /software/CESAR2.0-master/extra/tables/human/firstCodon_profile.txt /software/CESAR2.0-master/extra/tables/human/lastCodon_profile.txt --max-memory 10 > /dev/shm/cesarOut.aln.PsbKqi' failed".

**Comment by MichaelHiller** [comment]
Pls try to increase memory. This could be an problem. 

Otherwise, if you send me the input files and the exact command that you call, I recapitulate the problem. 
Alternatively, pls try TOGA.

**Comment by BoBo2021jenny** [comment]
Hi
Thank you for your reply！
Does this software consider issues such as premature termination codons and frameshifts that may arise from multiple exons becoming a coding sequence?

**Comment by MichaelHiller** [comment]
Yes, TOGA handles all that. It will recognize stop codons and frameshifts (listed in the inact_mutation file), uses them to classify the transcripts and it will mask them in the codon alignments.

**Comment by BoBo2021jenny** [comment]
Does Cesar 2.0 have this functionality?

**Comment by MichaelHiller** [comment]
no, only TOGA
